daemon/cmd/status.go

```diff
@@ -793,6 +795,48 @@ func (d *Daemon) startStatusCollector(cleaner *daemonCleanup) {
 				}
 			},
 		},
+		{
+			Name: "config-settings",
+			Probe: func(ctx context.Context) (interface{}, error) {
+				if !d.clientset.IsEnabled() || option.Config == nil {
```

Comment:
I don't think you should be querying Kubernetes for this, you should be reading the configuration from the filesystem. It's more accurate to be looking at the configuration Cilium would see if it was to restart. Also, kubelet polls for configMap changes, so it will skew for however long that poll interval is, which can be configured to be quite long, so this could unnecessarily report skew when it's expected.

Reply:
@chancez  Thank you for the review! Sounds like I am missing somethings...

From what I can tell from the code and setup the tmp volume type is emptyDir and populated via an init container using `cilium-dbg build-config`. Thus if the agent restarts the volume will be repopulated from the current configmap settings as the init-container will rebuild the cm dir struct from a direct call to the api server and hence picking up the latest no matter what the kubelet settings might be.  Which seems to me a more reliable signal??

Thus I think the check here make sense so users can see deltas directly from the api server regardless of the kubelet setting once the mount is in place from `configMapAndSecretChangeDetectionStrategy` which could potentially be stale as these settings affect how the mount is maintained afterwards.

Also if the cluster is scaled up, we may run into situations where some agent run newer settings than others as the volume mount will be hydrated from the api server directly.

Reply:
> cilium-dbg build-config

Ohhh right, that's related to the per-agent configuration changes I think, I didn't realize that. That complicates things even more, since your currently maintaining the deltas, but it's not accounting for the `build-config` logic, is it?

Reply:
TDH I am not fully versed on the whole config loading. I am sure there is more we can do here since the deltas are cilium-config cm specific. I think it's the straighter shot there as this should already add value to our customers/SAs knowing about potential out of date conditions especially in upgrade situations.

Reply:
So from what I understand when an agent starts an init container is run to gather configurations and dump them to disk as kv files. I uses an emptyDir volume and NOT a cm volume.  The agent then reads the volume and populates its configuration settings. If/When a user updates Cilium, there is a chance the configuration read from disk differs from the desired configuration and this is what we currently flag. This is very pragmatic ie if current state != desired state -> flag it. So not sure why this would lead users astray?? 
Am I missing it?

Reply:
Right, your not reading from disk though, your reading from the configmap, so your missing all the stuff cilium-dbg is doing, such as applying per-node configuration.

Reply:
#27972 stipulates reading the cm and checking for deltas.

Reply:
Ok if we want to focus solely on the configMap, let's at least make sure nothing is being missed. 

Based on https://github.com/cilium/cilium/pull/22656 it reads from the configMap, but also the `CiliumNodeConfig`. So https://github.com/cilium/cilium/blob/main/cilium-dbg/cmd/build-config.go#L90-L158 is the logic and it uses https://github.com/cilium/cilium/blob/main/pkg/option/resolver/resolver.go#L50. Overall it mostly looks like it merges the configurations.

So in the future, we can add labels to identify what source the configuration drift is coming from. The main thing to consider is that without this extra handling, anyone using `CiliumNodeConfig` will potentially get a false warning about the drift for those nodes if their per-node config overrides something in their cilium-config.

Reply:
Thank you for the clarification. Good point! I've added the necessary logic to account for potential per node config crd settings prior to computing the deltas.